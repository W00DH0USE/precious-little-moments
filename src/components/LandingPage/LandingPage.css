h1 {
  font-size: 45px;
}

h2 {
  font-size: 30px;
  font-weight: 500
}

h3 {
  font-size: 30px;
}

p {
  font-size: 21px;
  font-weight: 400;
  color: #333;
}

.LandingPage__App {
  margin-top: -30px !important;
}

.LandingPage__container {
  text-align: center;
}

.LandingPage__container {
  animation: moveInBottom ease-out .75s;
}

.LandingPage__banner {
  background-color: #fff;
  height: calc(100vh - 200px);
}

.heading {
  font-family: "Times New Roman",Georgia,Times,serif;
  font-weight: 500;
  line-height: 1.1;
  padding-top: 70px;
  margin: 0 0.67em 0;
}

.LandingPage-hr {
  background: linear-gradient(to right,rgb(0, 0, 0), rgb(97, 97, 97), rgb(0, 0, 0));
  height: 1px;
  border: none;
  width: 40%;
}

.LandingPage__Tagline {
  width: 80%;
  margin: auto auto;
}

.demo-button{
  width: 100px;
  font-size: 25px;
  margin-top: 15px;
  transition: .3s;
  background-color: #ee432e;
  border-color: #ec2e16;
  color: #fff;
}

.demo-button:hover {
  transform: translateY(-5px) scale(1.1);
  cursor: pointer;
}

.bgimage {
  background-image: url(https://new.thelittlememory.com/assets/images/leather-6f6902d873a5b06d39660840e6e22162.png);
  text-align: left;
  padding-top: 2em;
}

.features {
  align-items: center;
}

.container {
  margin-right: auto;
  margin-left: auto;
}

.HomePage__Footer {
  /* background-image: url(https://new.thelittlememory.com/assets/images/leather-6f6902d873a5b06d39660840e6e22162.png); */
}

.features {
  padding-top: 1.6em;
  padding-bottom: 2em;
  text-align: center;
}

.col-top {
  width: 100%;
  display: block;
}

.col-right-alt {
  display: none;
}

.col-left h2 {
  font-size: 30px;
  margin-top: 20px;
  margin-bottom: 10px;
  font-family: "Times New Roman",Georgia,Times,serif;
  font-weight: 500;
  line-height: 1.1;
}

.col-left p {
  margin: 0 0 10px;
  line-height: 1.42857;
  color: #333;
}

.joinNow-button {
  margin-top: 0;
  width: 150px;
}

.img-responsive1,
.img-responsive2 {
  display: block;
  margin-left: auto;
  margin-right: auto;
  max-width: 100%;
  height: auto;
}

.img-responsive1 {
  max-width: auto;
  height: 250px;
}

.col-left {
  width: 50%;
  min-height: 1px;
  margin:  20px auto;
}

.col-right {
  left: 8.33333%;
  width: 41.66667%;
  min-height: 1px;
  padding-left: 15px;
  padding-right: 15px;
  margin: 0 auto;
}

.HomePage__Footer {
  text-align: center;
}

.HomePage__Footer__Text {
  font-weight: 600;
  margin-top: -11px;
  padding-top: 80px;
}

.HomePage__Footer__Text a {
  text-decoration: none;
  color: #ec2e16;
  transition: all .4s;
}

.HomePage__Footer__Text a:hover {
  color: #0e76a8;
}

.HomePage__Footer__Icon {
  margin: 0 5px;
  font-size: 160%;
  transition: all .4s;
  color: #ec2e16;
}

.Linkedin__Icon:hover {
  color: #0e76a8;
}

.Github__Icon:hover {
  color: #000;
}

.HomePage__Footer__Copyright {
  font-size: 80%;
  margin-bottom: 0;
  padding-bottom: 20px;
}

/* Tooltip text */
.demo-button .tooltiptext {
  visibility: hidden;
  width: 100%;
  background-color: #555;
  color: #fff;
  text-align: center;
  padding: 5px 0;
  border-radius: 6px;
  font-size: 40%;

  /* Position the tooltip text */
  position: absolute;
  z-index: 1;
  top: 135%;
  left: 0%;

  /* Fade in tooltip */
  opacity: 0;
  transition: opacity 0.3s;
}

/* Tooltip arrow */
.demo-button .tooltiptext::after {
  content: "";
  position: absolute;
  bottom: 100%;
  left: 50%;
  margin-left: -5px;
  border-width: 5px;
  border-style: solid;
  border-color: #555 transparent transparent transparent;
  transform: rotate(180deg);
}

/* Show the tooltip text when you mouse over the tooltip container */
.demo-button:hover .tooltiptext {
  visibility: visible;
  opacity: 1;
}

@media only screen and (min-width: 768px) {

  login-button, reg-button {
    display: inline-block;
    width: 100px;
    margin: 50px 25px;
    border-radius: 5px;
    padding: 20px;
    font-size: 25px;
  } 
  .container {
    max-width: 730px;
  }
  .LandingPage__banner {
    background-image: url("../../images/banner.jpg");
  }
  .LandingPage__Tagline {
    width: 45%;
    margin: auto auto;
  }
  .demo-button{
    width: 200px;
    font-size: 25px;
    margin-top: 45px;
  }
  .col-right-alt {
    display: inherit;
  }
  .col-top {
    display: none;
  }
  .features {
    display: flex;
  }
  .col-left {
    float: left;
    padding-left: 15px;
    padding-right: 15px;
    margin-left: 0;
    margin-right: 0;
  }
  .features {
    text-align: inherit;
  }
  .bgimage {
    padding-top: 6em;
  }
}

@media only screen and (min-width: 1024px) {

  .features p {
    margin: 10px auto;
  }

  .container {
    width: 1170px;
  }

  .heading {
    padding-top: 200px;
    margin: 0.67em 0;
  }
}