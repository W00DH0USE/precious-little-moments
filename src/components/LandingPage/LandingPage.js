import React from 'react'
import { Button } from "../Utils/Utils";
import { Link } from 'react-router-dom'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import {  faGithub, faLinkedin  } from '@fortawesome/free-brands-svg-icons'
import './LandingPage.css'

function LandingPage(props) {
  
  return (
    <div className="LandingPage__app" style={{ marginTop: "-30px" }}>
      <div className="LandingPage__container">

        <div className="LandingPage__banner">
          <h1 className="heading">Precious Little Moments</h1>
          <hr className="LandingPage-hr"></hr>
          <section className="LandingPage__Tagline">
            <p>
            An online diary for parents who want to save and share those precious moments of your little ones.
            </p>
          </section>
          {!props.isLoading ? 
            <Button className="button demo-button" onClick={props.handleDemoLogin} style={{ width: "200px" }}>Demo Login</Button> 
            :
            <Button className="button demo-button" disabled={props.isLoading} style={{ width: "200px" }}>Loading...</Button>
          }
        </div>

        <div className="container-fluid bgimage">
          <div className="container features">
            <div className="col-top">
              <img src={require("../../images/smartmockups_k37xvb9w.png")} alt="screenshot-1" className="img-responsive1" />
            </div>
            
            <div className="col-left">
              <h2>Made Simple</h2>
              <p style={{ fontSize: "1.2em", marginBottom: "1em" }}>Join now and start saving those precious moments. It's that easy.</p>
              <Link to="/register">
                <Button className="button demo-button joinNow-button">Join Now</Button>
              </Link>
            </div>

            <div className="col-right col-right-alt">
              <img src={require("../../images/smartmockups_k37xvb9w.png")} alt="screenshot-1" className="img-responsive1" />
            </div>
          </div>
        </div>

        <div className="container-fluid bgimage">
          <div className="container features">
            <div className="col-right">
              <img src={require("../../images/smartmockups_k37xwkjj.png")} alt="screenshot-1" className="img-responsive2" />
            </div>

            <div className="col-left">
              <h2>Create Those Moments</h2>
              <p style={{ fontSize: "1.2em", marginBottom: "1em" }}>After you've registered and logged in, Get taken to your personnel profile where you will find all of your Moments. Add, Edit and Delete as you will. It's REALLY that easy!</p>
              <Link to="/login">
                <Button className="button demo-button joinNow-button">Login Now</Button>
              </Link>
            </div>

          </div>
        </div>

      </div>
      <footer className='HomePage__Footer'>
        <p className='HomePage__Footer__Text'>
          Created by
          <a href='http://www.iwoodhouse.net/' target='_blank' rel="noopener noreferrer"> Iyron Woodhouse</a> 
        </p>
        <div className='footer__social-media'>
          <a href="https://github.com/W00DH0USE" target="_blank" rel="noopener noreferrer">
            <FontAwesomeIcon className='gold HomePage__Footer__Icon Github__Icon' icon={faGithub} />
          </a>
          <a href="https://www.linkedin.com/in/iyron-woodhouse-49973b122/" target="_blank" rel="noopener noreferrer">
            <FontAwesomeIcon className='gold HomePage__Footer__Icon Linkedin__Icon' icon={faLinkedin} />
          </a>
        </div>
        <p className='HomePage__Footer__Copyright'>Copyright Â© 2019 - All Rights Reserved</p>
      </footer>
    </div>
  );
}

export default LandingPage


