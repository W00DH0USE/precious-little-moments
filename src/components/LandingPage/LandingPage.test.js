import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import LandingPage from './LandingPage';


describe(`Landing Page Component`, () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<LandingPage />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});