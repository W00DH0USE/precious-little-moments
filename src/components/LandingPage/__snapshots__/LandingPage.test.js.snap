// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Page Component renders without crashing 1`] = `
<div
  className="LandingPage__app"
  style={
    Object {
      "marginTop": "-30px",
    }
  }
>
  <div
    className="LandingPage__container"
  >
    <div
      className="LandingPage__banner"
    >
      <h1
        className="heading"
      >
        Precious Little Moments
      </h1>
      <hr
        className="LandingPage-hr"
      />
      <section
        className="LandingPage__Tagline"
      >
        <p>
          An online diary for parents who want to save and share those precious moments of your little ones.
        </p>
      </section>
      <Link
        to="/demo"
      >
        <Button
          className="button demo-button"
        >
          Tour
        </Button>
      </Link>
    </div>
    <div
      className="container-fluid bgimage"
    >
      <div
        className="container features"
      >
        <div
          className="col-top"
        >
          <img
            alt="screenshot-1"
            className="img-responsive1"
            src="smartmockups_k37xvb9w.png"
          />
        </div>
        <div
          className="col-left"
        >
          <h2>
            Made Simple
          </h2>
          <p
            style={
              Object {
                "fontSize": "1.2em",
                "marginBottom": "1em",
              }
            }
          >
            Join now and start saving those precious moments. It's that easy.
          </p>
          <Link
            to="/register"
          >
            <Button
              className="button demo-button joinNow-button"
            >
              Join Now
            </Button>
          </Link>
        </div>
        <div
          className="col-right col-right-alt"
        >
          <img
            alt="screenshot-1"
            className="img-responsive1"
            src="smartmockups_k37xvb9w.png"
          />
        </div>
      </div>
    </div>
    <div
      className="container-fluid bgimage"
    >
      <div
        className="container features"
      >
        <div
          className="col-right"
        >
          <img
            alt="screenshot-1"
            className="img-responsive2"
            src="smartmockups_k37xwkjj.png"
          />
        </div>
        <div
          className="col-left"
        >
          <h2>
            Create Those Moments
          </h2>
          <p
            style={
              Object {
                "fontSize": "1.2em",
                "marginBottom": "1em",
              }
            }
          >
            After you've registered and login. Get taken to your personnel dashboard where you will find all of your Moments. Add, Edit and Delete as you will. It's REALLY that easy!
          </p>
          <Link
            to="/login"
          >
            <Button
              className="button demo-button joinNow-button"
            >
              Login Now
            </Button>
          </Link>
        </div>
      </div>
    </div>
  </div>
  <footer
    className="HomePage__Footer"
  >
    <p
      className="HomePage__Footer__Text"
    >
      Created by
      <a
        href="http://www.iwoodhouse.net/"
        rel="noopener noreferrer"
        target="_blank"
      >
         Iyron Woodhouse
      </a>
    </p>
    <div
      className="footer__social-media"
    >
      <a
        href="https://github.com/W00DH0USE"
        rel="noopener noreferrer"
        target="_blank"
      >
        <FontAwesomeIcon
          border={false}
          className="gold HomePage__Footer__Icon Github__Icon"
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                496,
                512,
                Array [],
                "f09b",
                "M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z",
              ],
              "iconName": "github",
              "prefix": "fab",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </a>
      <a
        href="https://www.linkedin.com/in/iyron-woodhouse-49973b122/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <FontAwesomeIcon
          border={false}
          className="gold HomePage__Footer__Icon Linkedin__Icon"
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                448,
                512,
                Array [],
                "f08c",
                "M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z",
              ],
              "iconName": "linkedin",
              "prefix": "fab",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </a>
    </div>
    <p
      className="HomePage__Footer__Copyright"
    >
      Copyright Â© 2019 - All Rights Reserved
    </p>
  </footer>
</div>
`;
